### PR TITLE
Feat/593 adaptation run a cost analysis

### PR DESCRIPTION
**Pull request by ArdtK**
## Issue addressed
Solves #593

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [x] I HAVE discussed my solution with (other) members of the RA2CE team.

## What has been done?
- Add a minimal `Adaptation` class with tests.
- Add method `calc_option_cost` to `AdaptationOption`

### Checklist
- [x] Code is formatted using our custom `black` and `isort` definitions.
- [x] Tests are either added or updated.
- [x] Branch is up to date with `master`.
- [ ] Updated documentation if needed.

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: ra2ce/analysis/adaptation/adaptation.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            analysis_config
+        )
+
+    def execute(self) -> GeoDataFrame | None:
+        """
+        Run the adaptation analysis.
+        """
+        return self.calculate_bc_ratio()
+
+    def calculate_option_cost(self, option: AdaptationOption) -> float:
```
Comment (MatthiasHauthDeltares) [inline ra2ce/analysis/adaptation/adaptation.py:61]: Shouldn't  it a method of the class AdaptationOption?

File: ra2ce/analysis/adaptation/adaptation.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                option.maintenance_interval,
+            )
+            for _maint_year in _maint_years:
+                _lifetime_cost += calc_cost(option.maintenance_cost, _maint_year)
+
+        return _lifetime_cost
+
+    def run_cost(self) -> GeoDataFrame | None:
+        """
+        Calculate the cost of the adaptation.
```
Comment (MatthiasHauthDeltares) [inline ra2ce/analysis/adaptation/adaptation.py:95]: Calculate the cost of the adaptation, for all adaptation options in collection.

File: ra2ce/analysis/adaptation/adaptation.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        """
+        return 0.0
+
+    def run_benefit(self) -> GeoDataFrame | None:
+        """
+        Calculate the benefit of the adaptation.
```
Comment (MatthiasHauthDeltares) [inline ra2ce/analysis/adaptation/adaptation.py:101]: same as above

File: ra2ce/analysis/adaptation/adaptation.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        """
+        return 0.0
+
+    def calculate_bc_ratio(self) -> GeoDataFrame | None:
+        """
+        Calculate the benefit-cost ratio of the adaptation.
```
Comment (MatthiasHauthDeltares) [inline ra2ce/analysis/adaptation/adaptation.py:107]: same as above

File: ra2ce/analysis/adaptation/adaptation.py (hunk 1 above)
Comment (MatthiasHauthDeltares) [reply ra2ce/analysis/adaptation/adaptation.py:61]: Add in docstrings: calculate the net value unit cost (per m)

**Review by MatthiasHauthDeltares — CHANGES_REQUESTED** [review]

File: ra2ce/analysis/adaptation/adaptation.py (hunk 2 above)
Comment (ArdtK) [reply ra2ce/analysis/adaptation/adaptation.py:95]: Agree, that is for the next PR.

File: ra2ce/analysis/adaptation/adaptation.py (hunk 3 above)
Comment (ArdtK) [reply ra2ce/analysis/adaptation/adaptation.py:101]: Agree, that is for a next PR

File: ra2ce/analysis/adaptation/adaptation.py (hunk 1 above)
Comment (ArdtK) [reply ra2ce/analysis/adaptation/adaptation.py:61]: Fair point, will adjustust

**Review by MatthiasHauthDeltares — APPROVED** [review]

**Pull request merged**